pkg/transport/udpmultimux_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transport
+
+import (
+	"net"
+	"testing"
+
+	"github.com/pion/ice/v4"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/slices"
+)
+
+type mockUDPMux struct {
+	addrs []net.Addr
+}
+
+func (m *mockUDPMux) GetListenAddresses() []net.Addr {
+	return m.addrs
+}
+
+func (m *mockUDPMux) GetConn(ufrag string, addr net.Addr) (net.PacketConn, error) {
+	return nil, nil
+}
+
+func (m *mockUDPMux) RemoveConnByUfrag(ufrag string) {}
+
+func (m *mockUDPMux) Close() error {
+	return nil
+}
+
+func TestMultiPortsUDPMux(t *testing.T) {
+	mux1 := &mockUDPMux{
+		addrs: []net.Addr{
+			&net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: 8000},
+		},
+	}
+
+	mux2 := &mockUDPMux{
+		addrs: []net.Addr{
+			&net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: 8001},
+		},
+	}
+
+	mux3 := &mockUDPMux{
+		addrs: []net.Addr{
+			&net.UDPAddr{IP: net.ParseIP("192.168.1.100"), Port: 8000},
+		},
+	}
+
+	// Create a mux with different IP
```

Comment:
:nit: should this comment be above `mux3` as `mux3` is using a different IP already?